### PR TITLE
Aliaks nad/back end

### PR DESCRIPTION
**Pull request by anadtochaev**
+ was created DBcontext for game logic
* was changed game service and controller with implementation functional pattern

### CODE REVIEW

File: GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs [hunk 1]
```diff
@@ -34,4 +30,24 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
             modelBuilder.Configurations.Add(new UserDbConfiguration());
         }
     }
+
+    public sealed class FleetContexts : DbContext
+    {
+        public FleetContexts(ILogger logger)
+        {
+            //Database.SetInitializer<UsersContexts>(new MigrateDatabaseToLatestVersion<UsersContexts, Configuration>());
+            Database.SetInitializer<FleetContexts>(new DropCreateDatabaseAlways<FleetContexts>());
```
Comment (belpyro) [inline GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs]: are you sure?

File: GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs [hunk 2]
```diff
@@ -34,4 +30,24 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
+            Database.Log = msg => logger.Warning(msg);
```
Comment (belpyro) [inline GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs]: warning lever or different?

File: GamePortal/AliaksNad.Battleship.Data/Migrations/202005261930481_InitialCreate.cs [hunk 3]
```diff
@@ -1,48 +1,48 @@
-﻿namespace AliaksNad.Battleship.Data.Migrations
-{
-    using System;
-    using System.Data.Entity.Migrations;
+﻿//namespace AliaksNad.Battleship.Data.Migrations
+//{
+//    using System;
+//    using System.Data.Entity.Migrations;
```
Comment (belpyro) [inline GamePortal/AliaksNad.Battleship.Data/Migrations/202005261930481_InitialCreate.cs]: don't commit commented code!

File: GamePortal/AliaksNad.Battleship.Logic/BattleshipLogicDIModule.cs [hunk 4]
```diff
@@ -17,12 +17,14 @@ public class BattleshipLogicDIModule : NinjectModule
     {
         public override void Load()
         {
-            Mapper.Initialize(cfg => cfg.AddProfiles(typeof(UserProfile)));
+            Mapper.Initialize(cfg => cfg.AddProfiles(typeof(UserProfile), typeof(FleetProfile), typeof(CoordinatesProfile)));
```
Comment (belpyro) [inline GamePortal/AliaksNad.Battleship.Logic/BattleshipLogicDIModule.cs]: don't use this approach! check my last commit

File: GamePortal/AliaksNad.Battleship.Logic/Models/UserDto.cs [hunk 5]
```diff
@@ -19,4 +19,20 @@ public class UserDto
 
         public IEnumerable<StatisticDto> Statistics { get; set; }
     }
+
+    public class FleetDto
```
Comment (belpyro) [inline GamePortal/AliaksNad.Battleship.Logic/Models/UserDto.cs]: one type - one file

**Review by belpyro — COMMENTED** [review]
Check your code and fix all my suggestions!

File: GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs (hunk 1 above)
Comment (anadtochaev) [reply GamePortal/AliaksNad.Battleship.Data/Contexts/UsersContexts.cs]: yes, right now i develop db and don't need to keep data

**Review by belpyro — CHANGES_REQUESTED** [review]
check commented code before merging

**Review by anadtochaev — COMMENTED** [review]
changed into actual commit

**Review by belpyro — APPROVED** [review]
try to select right type more accuracy

**Pull request merged**